ui/src/components/ContextMenu/index.tsx

```diff
@@ -1,201 +1,89 @@
-    </span>
-    {children}
-  </ContextMenuPrimitive.CheckboxItem>
-));
-ContextMenuCheckboxItem.displayName =
-  ContextMenuPrimitive.CheckboxItem.displayName;
+const ContextMenu: React.FC<ContextMenuProps> = ({
+  items,
+  onClose,
+  menuPosition,
+}) => {
+  const { top, left, right, bottom } = menuPosition;
+  return (
+    <div
+      className="absolute z-50"
+      style={{
+        top: top !== false ? top : undefined,
+        left: left !== false ? left : undefined,
+        right: right !== false ? right : undefined,
+        bottom: bottom !== false ? bottom : undefined,
+      }}>
+      <div className="min-w-[160px] select-none rounded-md border bg-card p-1 text-popover-foreground shadow-md">
+        {items.map((item, index) => (
+          <ContextMenuItem
+            key={index}
+            label={item.label}
+            icon={item.icon}
+            onCallback={item.onCallback}
+            onClose={onClose}
+            destructive={item.destructive}
+            disabled={item.disabled}
+          />
+        ))}
+      </div>{" "}
+    </div>
+  );
+};
 
-const ContextMenuRadioItem = React.forwardRef<
-  React.ElementRef<typeof ContextMenuPrimitive.RadioItem>,
-  React.ComponentPropsWithoutRef<typeof ContextMenuPrimitive.RadioItem>
->(({ className, children, ...props }, ref) => (
-  <ContextMenuPrimitive.RadioItem
-    ref={ref}
-    className={cn(
-      "relative flex cursor-pointer select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-hidden focus:bg-accent focus:text-accent-foreground data-disabled:pointer-events-none data-disabled:opacity-50",
-      className,
-    )}
-    {...props}>
-    <span className="absolute left-2 flex size-3.5 items-center justify-center">
-      <ContextMenuPrimitive.ItemIndicator>
-        <DotFilledIcon className="size-4 fill-current" />
-      </ContextMenuPrimitive.ItemIndicator>
-    </span>
-    {children}
-  </ContextMenuPrimitive.RadioItem>
-));
-ContextMenuRadioItem.displayName = ContextMenuPrimitive.RadioItem.displayName;
+type ContextMenuItemProps = {
+  label: string;
+  icon?: React.ReactNode;
+  className?: string;
+  onCallback: () => void;
+  onClose: () => void;
+  destructive?: boolean;
+  disabled?: boolean;
+};
 
-const ContextMenuLabel = React.forwardRef<
-  React.ElementRef<typeof ContextMenuPrimitive.Label>,
-  React.ComponentPropsWithoutRef<typeof ContextMenuPrimitive.Label> & {
-    inset?: boolean;
-  }
->(({ className, inset, ...props }, ref) => (
-  <ContextMenuPrimitive.Label
-    ref={ref}
-    className={cn(
-      "px-2 py-1.5 text-sm font-semibold text-foreground",
-      inset && "pl-8",
-      className,
-    )}
-    {...props}
-  />
-));
-ContextMenuLabel.displayName = ContextMenuPrimitive.Label.displayName;
+const ContextMenuItem: React.FC<ContextMenuItemProps> = ({
+  label,
+  icon,
+  className,
+  destructive,
+  disabled,
+  onCallback,
+  onClose,
+}) => {
+  const isDescructive = destructive ? "text-destructive" : "";
```

Comment:
I know you don't like it, but please just have this and isDisabled in the className